src/index.js

```diff
@@ -161,12 +172,9 @@ const printers = {
   },
 };
 
-module.exports = {
-  languages,
-  printers,
-  parsers,
-  options,
-  defaultOptions: {
-    tabWidth: 4,
-  },
+const defaultOptions = {
+  tabWidth: 4,
 };
+
+export { languages, printers, parsers, options, defaultOptions };
+export default { languages, printers, parsers, options, defaultOptions };
```

Comment:
I would strongly feel that only one canonic way of importing the plugin should be offered (either `import PhpPlugin from '…'` or `import * as PhpPlugin from '…'`, not both).

Prettier's built-in plugins transitioned to multiple exports (e.g. `import * as HtmlPlugin from 'prettier/plugins/html'`, so that's what I would go with.

Since v0.20 yesterday was a completely breaking change anyway (by dropping Prettier v1/v2 support), I also don't see any BC related benefits from keeping the default export around.

Or am I missing something here? Opinions, @czosel?

Reply:
> Prettier's built-in plugins transitioned to multiple exports

Actually this is not true, I'm the one add all default export to them, check [this esbuild plugin](https://github.com/prettier/prettier/blob/main/scripts/build/esbuild-plugins/add-default-export.js).

I added default export because Prettier v3 made lots of breaking changes, I don't want to force users already uses default import to import Prettier v2 change their code. And I plan to remove the extra default export in Prettier v4. That's the same reason I added default export here.

Anyway, I agree with your opinion, and I'm fine to remove the default export. [`28b8c9e` (#2213)](https://github.com/prettier/plugin-php/pull/2213/commits/28b8c9ee8434d065f26b8e72d45bea66cbba9956)

Reply:
Ah, good to know, thanks. I only took a look at the source code of the plugins, sorry.
Then of course I won't object, streamlining the plugin APIs sounds just as valuable as having one true way™ of doing things. Do as you deem appropriate. :)